bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java

```diff
@@ -238,14 +247,14 @@ private void initialAccess(BoschHttpClient httpClient) {
 
     /**
      * Get a list of connected devices from the Smart-Home Controller
-     * 
+     *
      * @throws InterruptedException in case bridge is stopped
      */
-    private boolean getDevices() throws InterruptedException {
+    public @Nullable ArrayList<Device> getDevices() throws InterruptedException {
         @Nullable
         BoschHttpClient httpClient = this.httpClient;
         if (httpClient == null) {
-            return false;
+            return null;
```

Comment:
In my opinion we should start to throw something like a NotInitializedException in cases like this. What do you think?

Reply:
- [ ] throw exception instead return null - will also avoid @Nullable